pkg/frontend/mysql_protocol.go

```diff
@@ -1631,6 +1631,127 @@ func (mp *MysqlProtocolImpl) sendColumns(mrs *MysqlResultSet, cmd int, warnings,
 	return nil
 }
 
+// the server convert every row of the result set into the format that mysql protocol needs
+func (mp *MysqlProtocolImpl) makeResultSetBinaryRow(buffer []byte, mrs *MysqlResultSet, rowIdx uint64) ([]byte, error) {
+	buffer = append(buffer, defines.OKHeader) // append OkHeader
```

Comment:
remove the buffer. 
use mp.appendStringLenEnc which maintain the internal buffer for you.

Reply:
in makeResultSetTextRow , it use a lot of mp.appendXXX which maintain the network buffer.